### PR TITLE
[IMP] product_unique_serial: replaced check unicity method …

### PR DESCRIPTION
**Pull request by suniagajose**
Replaced previous check unicity method by a new one extracted from v9
## Features
- Now validation of 'Unique Lot' is from quants and not from move as was done previously.
- Validate unicity when consume products in manufacturing orders.
- Validate unicity when split a line and each one has same serial number.
- Fixed old TODO to recomputed Last Location in Lot model.
- Standardize and solved some pylint, flake8, pylint-odoo issues

### CODE REVIEW

**Comment by moylop260** [comment]
Good job!
Extracted from odoo/9.0
Fix red travis please

File: product_unique_serial/tests/test_for_unicity.py [hunk 1]
```diff
@@ -173,10 +173,7 @@ def test_1_1product_1serialnumber_2p_in(self):
         self.transfer_picking(
             picking_1,
             self.env.ref('product_unique_serial.serial_number_demo_1'))
-        with self.assertRaisesRegexp(
-                except_orm,
-                r"you will have a quantity of '2.0' "
-                r"in lot '86137801852514'"):
+        with self.assertRaises(exceptions.Warning):
```
Comment (moylop260) [inline product_unique_serial/tests/test_for_unicity.py]: @suniagajose ever is better use: `assertRaisesRegexp` to get the real traceback message

**Comment by moylop260** [comment]
@suniagajose 
Could you add the unittest for cover next lines? (Check red ones)
 https://coveralls.io/builds/4014103/source?filename=product_unique_serial%2Fmodel%2Fstock.py

**Comment by suniagajose** [comment]
sure!

2015-11-01 16:20 GMT-04:30 moylop260 notifications@github.com:

**Comment by moylop260** [comment]
@suniagajose 
This pr add a warning in runbot.
Could you fix it?

**Pull request merged**